interfaces/cython/cantera/thermo.pyx

```diff
@@ -1647,76 +1707,183 @@ cdef class PureFluid(ThermoPhase):
+            warnings.warn("To be removed after Cantera 2.5. "
+                          "Attribute renamed to 'TPQ'", DeprecationWarning)
+            self.TPQ = values
+
+    property TPQ:
+        """
+        Get/Set the temperature [K], pressure [Pa], and vapor fraction of a
+        PureFluid.
+
+        An Exception is raised if the thermodynamic state is not consistent.
+        """
+        def __get__(self):
+            return self.T, self.P, self.Q
         def __set__(self, values):
             T = values[0] if values[0] is not None else self.T
             P = values[1] if values[1] is not None else self.P
-            X = values[2] if values[2] is not None else self.X
-            if not isinstance(X, (np.ndarray, _numbers.Number)):
+            Q = values[2] if values[2] is not None else self.Q
+            if not isinstance(Q, (np.ndarray, _numbers.Number)):
                 raise ValueError(
                     'a numeric value is required to quanitify '
-                    'the vapor fraction (X)'
+                    'the vapor fraction (Q)'
                 )
             if np.isclose(P, self.thermo.satPressure(T)):
-                self.TX = T, X
-            elif np.isclose(X, 0.) or np.isclose(X, 1.):
+                self.TQ = T, Q
+            elif np.isclose(Q, 0.) or np.isclose(Q, 1.):
                 self.TP = T, P
             else:
                 raise ValueError(
-                    'invalid thermodynamic state: the TPX setter '
+                    'invalid thermodynamic state: the TPQ setter '
                     'received a combination of property values that '
                     'do not represent a valid state. As an alternative, '
                     'specify the state using two fully independent '
                     'properties (e.g. TD) instead of: '
-                    'T={}, P={}, X={}'.format(T, P, X)
+                    'T={}, P={}, Q={}'.format(T, P, Q)
                 )
 
     property UVX:
         """
         Get the internal energy [J/kg or J/kmol], specific volume
         [m^3/kg or m^3/kmol], and vapor fraction.
+
+        .. deprecated:: 2.5
+
+             To be deprecated with version 2.5, and removed thereafter.
+             Renamed to `UVQ`.
         """
         def __get__(self):
-            return self.u, self.v, self.X
+            warnings.warn("To be removed after Cantera 2.5. "
+                          "Attribute renamed to 'UVQ'", DeprecationWarning)
+            return self.UVQ
+
+    property UVQ:
+        """
+        Get the internal energy [J/kg or J/kmol], specific volume
+        [m^3/kg or m^3/kmol], and vapor fraction.
+        """
+        def __get__(self):
+            return self.u, self.v, self.Q
 
     property DPX:
+        """Get the density [kg/m^3], pressure [Pa], and vapor fraction.
+
+        .. deprecated:: 2.5
+
+             To be deprecated with version 2.5, and removed thereafter.
+             Renamed to `DPQ`.
```

Comment:
Would it be worth noting in these deprecation messages that the change in behavior after Cantera 2.5 won't be that this property doesn't exist, but that it will return the mole fraction instead of the vapor fraction? Likewise for the other properties that are shared with `ThermoPhase`.